### PR TITLE
test(pkg/context): migrate test-infra to testify for pkg/context pkg

### LINKED ISSUE #2859: migrate test-infra to testify for `pkg/context` pkg

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve? 
migrate test-infra to testify for pkg/context pkg，close issue [#2859](https://github.com/pingcap/ticdc/issues/2859), close issue [#2874](https://github.com/pingcap/ticdc/issues/2874)

### What is changed and how it works?

1. migrate test-infra from gocheck to testify
2. migrate leak-test from testleak to goleak

### Check List 

Tests 

 - Unit test

Code changes

 
### Release note 

```release-note
migrate test-infra to testify for pkg/context pkg
```

### CODE REVIEW

File: pkg/util/leak_helper.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package util
+
+import (
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+func IgnoreEtcdAndOpencensusLeak(m *testing.M) {
```
Comment (amyangfei) [inline pkg/util/leak_helper.go:22]: What about adding a parameters `...goleak.Option` to allow more customized ignore function.

**Comment by ben1009** [comment]
The pr description may need modify a bit. Rest lgtm

File: pkg/util/leak_helper.go (hunk 1 above)
Comment (maxshuang) [reply pkg/util/leak_helper.go:22]: Sure, i will change the function name also.

**Review by overvenus — APPROVED** [review]
LGTM

**Comment by maxshuang** [comment]
@ben1009 @3AceShowHand  PTLA , thanks.

**Review by ben1009 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Comment by maxshuang** [comment]
> /merge

will it automatically close the issue for me? Cause i mention the related issue in my pr comment.

**Pull request merged**